src/main/java/com/divudi/bean/common/SearchController.java

```diff
@@ -359,39 +360,55 @@ public String settleBillByBarcode() {
 
     }
 
-    public String toSettle(Bill args) {
-        String sql = "Select b from BilledBill b"
+    public String toSettle(Bill preBill) {
+        if (preBill == null) {
+            JsfUtil.addErrorMessage("No Such Prebill");
+            return "";
+        }
+        String sql = "Select b from "
+                + " BilledBill b"
                 + " where b.referenceBill=:bil"
                 + " and b.retired=false "
                 + " and b.cancelled=false ";
         HashMap hm = new HashMap();
-        hm.put("bil", args);
+        hm.put("bil", preBill);
         Bill b = getBillFacade().findFirstByJpql(sql, hm);
 
         if (b != null) {
             JsfUtil.addErrorMessage("Allready Paid");
             return "";
-        } else {
+        }
 
-            BillType btype = args.getBillType();
+        if (preBill.getBillTypeAtomic() != null) {
+            BillTypeAtomic bta = preBill.getBillTypeAtomic();
+            switch (bta) {
+                case OPD_BATCH_BILL_TO_COLLECT_PAYMENT_AT_CASHIER:
+                    return opdPreSettleController.toSettle(preBill);
+                default:
+                    System.out.println("No Adomic bill type for = " + b);
+            }
+        }
+        if (preBill.getBillType() != null) {
+            BillType btype = preBill.getBillType();
             switch (btype) {
                 case OpdPreBill:
-                    setPreBillForOpd(args);
+                    setPreBillForOpd(preBill);
                     return "/opd/opd_bill_pre_settle?faces-redirect=true";
 
                 case PharmacyPre:
-                    setPreBillForPharmecy(args);
+                    setPreBillForPharmecy(preBill);
                     return "/pharmacy/pharmacy_bill_pre_settle";
 

```

Comment:
Replace
`return "/pharmacy/pharmacy_bill_pre_settle";`
to 
`return "/pharmacy/pharmacy_bill_pre_settle?faces-redirect=true";`